sfdoc/publish/salesforce.py

```diff
@@ -296,19 +310,9 @@ def article_info_cache(self, publish_status, **filters):
         publish_status = publish_status.lower()
         key = (self.docset_uuid, publish_status)
         if not self._article_info_cache.get(key):
-            fields = ["Id", "KnowledgeArticleId", "Title", "Summary", "IsVisibleInCsp",
-                            "IsVisibleInPkb", "IsVisibleInPrm", "UrlName", "PublishStatus",
-                            settings.SALESFORCE_ARTICLE_BODY_FIELD,
-                            settings.SALESFORCE_ARTICLE_AUTHOR_FIELD,
-                            settings.SALESFORCE_ARTICLE_AUTHOR_OVERRIDE_FIELD,
-                            self.docset_uuid_join_field]
-            where_clauses = {"language": "en_US",
-                             "PublishStatus": publish_status}
-
-            if self.docset_scoped:
-                where_clauses[self.docset_uuid_join_field] = self.docset_uuid
-
-            self._article_info_cache[key] = self.query_articles(fields, where_clauses)
+            self._article_info_cache[key] = self._cache_population_query(publish_status)
+        if settings.CACHE_VALIDATION_MODE:
+            assert self._article_info_cache[key] == self._cache_population_query(publish_status)
```

Comment:
I don't understand the value of comparing the cached value to the actual query result immediately after setting the cached value to the actual query result. Is the `if settings.CACHE_VALIDATION_MODE` line supposed to be an `elif`?

Reply:
You're right. It's redundant in the cache-miss case so it would be a bit more efficient and clear to have it be an elif. I'll do that.

Reply:
7c3926fd17316281e664ce36f76069ae95dce146